gvm/protocols/gmpv7.py

```diff
@@ -1499,35 +1499,288 @@ def help(self, **kwargs):
         cmd = self._generator.help_command(kwargs)
         return self.send_command(cmd)
 
-    def modify_agent(self, agent_id, name='', comment=''):
-        cmd = self._generator.modify_agent_command(agent_id, name, comment)
-        return self.send_command(cmd)
+    def modify_agent(self, agent_id, name=None, comment=None):
+        """Generates xml string for modify agent on gvmd
 
-    def modify_alert(self, alert_id, **kwargs):
-        cmd = self._generator.modify_alert_command(alert_id, kwargs)
-        return self.send_command(cmd)
+        Arguments:
+            agent_id (int) ID of the agent to be modified.
+            name (str, optional): Name of the new credential
+            comment (str, optional): Comment for the credential
+        """
+        if not agent_id:
+            raise RequiredArgument('modify_agent requires agent_id argument')
+
+        cmd = XmlCommand('modify_agent')
+        cmd.set_attribute('agent_id', str(agent_id))
+        if name:
+            cmd.add_element('name', name)
+        if comment:
+            cmd.add_element('comment', comment)
+
+        return self._send_xml_command(cmd)
+
+    def modify_alert(self, alert_id, name=None, comment=None,
+                             filter_id=None, event= None, condition=None,
+                             method=None):
+        """Generates xml string for modify alert on gvmd.
+
+        Arguments:
+            alert_id (int) ID of the alert to be modified.
+            name (str): Name of the Alert.
+            condition (str): The condition that must be satisfied for the alert
+                to occur.
+            event (str): The event that must happen for the alert to occur
+            method (str): The method by which the user is alerted
+            filter_id (str, optional): Filter to apply when executing alert
+            comment (str, optional): Comment for the alert
+        """
+
+        if not alert_id:
+            raise RequiredArgument('modify_alert requires an alert_id argument')
+
+        cmd = XmlCommand('modify_alert')
+        cmd.set_attribute('alert_id', str(alert_id))
+
+        if name:
+            cmd.add_element('name', name)
+
+        if comment:
+            cmd.add_element('comment', comment)
+
+        if filter_id:
+            cmd.add_element('filter', attrs={'id': filter_id})
+
+        if len(event) > 1:
+            _xmlevent = cmd.add_element('event', event[0])
```

Comment:
Could you please split the event into two parameters (event and event_data) like in create_alert